client/src/main/java/com/networknt/client/ssl/Light4jALPNClientSelector.java

```diff
@@ -0,0 +1,154 @@
+package com.networknt.client.ssl;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.net.ssl.SSLEngine;
+
+import org.xnio.ChannelListener;
+import org.xnio.channels.StreamSourceChannel;
+import org.xnio.conduits.PushBackStreamSourceConduit;
+import org.xnio.ssl.SslConnection;
+
+import com.networknt.client.Http2Client;
+import com.networknt.config.Config;
+
+import io.undertow.client.ALPNClientSelector;
+import io.undertow.client.ALPNClientSelector.ALPNProtocol;
+import io.undertow.client.ClientCallback;
+import io.undertow.client.ClientConnection;
+import io.undertow.protocols.alpn.ALPNManager;
+import io.undertow.protocols.alpn.ALPNProvider;
+import io.undertow.protocols.ssl.SslConduit;
+import io.undertow.protocols.ssl.UndertowXnioSsl;
+import io.undertow.util.ImmediatePooled;
+
+/**
+ * Customized ALPNClientSelector for handling TLS handshake for HTTP2.
+ * 
+ * @author Daniel Zhao
+ *
+ */
+public class Light4jALPNClientSelector {
+	@SuppressWarnings("unchecked")
+	private static final Map<String, Object> TLS_CONFIG = (Map<String, Object>)Config.getInstance().getJsonMapConfig(Http2Client.CONFIG_NAME).get(Http2Client.TLS);
+	
+    public static void runAlpn(final SslConnection sslConnection, final ChannelListener<SslConnection> fallback, final ClientCallback<ClientConnection> failedListener, final ALPNProtocol... details) {
+        SslConduit conduit = UndertowXnioSsl.getSslConduit(sslConnection);
+        SSLEngine engine = UndertowXnioSsl.getSslEngine(sslConnection);
+        
```

Comment:
For me this method would definitely need some explanation

Reply:
I'll add a short description.